src/parser.rs

```diff
@@ -4746,16 +4746,57 @@ impl<'a> Parser<'a> {
                         self.expect_keyword(Keyword::JOIN)?;
                         JoinOperator::Inner
                     }
-                    kw @ Keyword::LEFT | kw @ Keyword::RIGHT | kw @ Keyword::FULL => {
+                    kw @ Keyword::LEFT | kw @ Keyword::RIGHT => {
+                        let _ = self.next_token();
+                        let join_type = self.parse_one_of_keywords(&[
+                            Keyword::OUTER,
+                            Keyword::SEMI,
+                            Keyword::ANTI,
+                            Keyword::JOIN,
+                        ]);
+                        match join_type {
+                            Some(Keyword::OUTER) => {
+                                self.expect_keyword(Keyword::JOIN)?;
+                                match kw {
+                                    Keyword::LEFT => JoinOperator::LeftOuter,
+                                    Keyword::RIGHT => JoinOperator::RightOuter,
+                                    _ => unreachable!(),
```

Comment:
@mingmwang for this and all the next `unreachable!()`. Even if this is really unreachable, please try to add an error here, or handle it in a different way. Being unsafe is not really good.

Reply:
Since it can have only two values, we can also make it a boolean value. I plan to merge this PR and then I will submit a follow on to remove the use of `unreachable!`